src/Param.cc

```diff
@@ -359,6 +360,63 @@ bool ParseUsingStringStream(const std::string &_input, const std::string &_key,
   return true;
 }
 
+//////////////////////////////////////////////////
+/// \brief Helper function for Param::ValueFromString for parsing colors
+/// This checks the color components specified in _input are rgb or rgba
+/// (expects 3 or 4 values) and each value is between [0,1]. When only 3 values
+/// are present (rgb), then alpha is set to 1.
+/// \param[in] _input Input string.
+/// \param[in] _key Key of the parameter, used for error message.
+/// \param[out] _value This will be set with the parsed value.
+/// \return True if parsing colors succeeded.
+bool ParseColorUsingStringStream(const std::string &_input,
+    const std::string &_key, ParamPrivate::ParamVariant &_value)
+{
+  StringStreamClassicLocale ss(_input);
+  std::string token;
+  std::vector<float> colors;
+  float c;  // r,g,b,a values
+  bool isValidColor = true;
+  while (ss >> token)
+  {
+    try
+    {
+      c = std::stof(token);
+      colors.push_back(c);
+    }
+    catch(const std::exception &/*e*/)
```

Comment:
Can the caught exception be more specific like the ones in `ValueFromString`

https://github.com/osrf/sdformat/blob/deb142130c6fb96b24a2e0f35e481e34ac014834/src/Param.cc#L509-L520

Reply:
Yup! dbf761434d51f8bc7bdc514a99cfc46439b304e3